// eil-common/build/src/buildLogger/buildLogger.client.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = buildLogger;

var _logLevel = require('../helpers/logLevel');

var _logLevel2 = _interopRequireDefault(_logLevel);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var logLevels = {
  fatal: 60,
  error: 50,
  warn: 40,
  info: 30,
  debug: 20,
  trace: 10
};
function buildLogger(_ref) {
  var name = _ref.name,
      appProperties = _ref.appProperties;

  var env = appProperties.get('env') || "development";
  var configLogLevel = appProperties.get('logger.level');
  var level = configLogLevel && configLogLevel !== 'env' ? configLogLevel : (0, _logLevel2.default)(env);
  var levelNum = logLevels[level];

  return Object.keys(logLevels).reduce(function (logger, key) {
    logger[key] = function () {
      for (var _len = arguments.length, messages = Array(_len), _key = 0; _key < _len; _key++) {
        messages[_key] = arguments[_key];
      }

      if (logLevels[key] >= levelNum) {
        var _console;

        // eslint-disable-next-line no-console
        (_console = console).log.apply(_console, ['[' + key.toUpperCase() + ': ' + name + ']'].concat(messages));
      }
    };
    return logger;
  }, {});
}
